zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/internal/client/HttpCall.java

```diff
@@ -86,11 +86,14 @@ public Factory(HttpClient httpClient) {
       if (t != null) {
         callback.onError(t);
       } else {
+        V value = null;
         try {
-          callback.onSuccess(parseResponse(response, bodyConverter));
-        } catch (IOException e) {
-          callback.onError(e);
+          value = parseResponse(response, bodyConverter);
+        } catch (Throwable t1) {
+          propagateIfFatal(t1);
+          callback.onError(t1);
         }
+        if (value != null) callback.onSuccess(value);
```

Comment:
Think this needs to go right below line 91. `parseResponse` can return `null` (line 133), which will cause this future to never complete.